review/rolling.js

```diff
@@ -0,0 +1,58 @@
+import { CLASS_LIST } from '../util/data';
+import { li } from '../util/htmlTemplate';
+import { delay } from '../util/util.js';
+
+function RecommendRolling({ data, selector, animation }) {
+  this.rollingData = data;
+  this.rollingList = selector.rollingList;
+  this.searchInput = selector.searchInput;
+  this.step = 0;
+  this.oneStep = animation.oneStep;
+  this.transition = animation.transition;
+}
+
+RecommendRolling.prototype = {
+  constructor: RecommendRolling,
+  init() {
+    this.render();
+    this.autoRolling();
+  },
+  async autoRolling() {
+    if (this.step > this.rollingData.length) {
+      this.render();
+      this.step = 0;
+      await delay('', 0);
+      this.rolling();
+      this.autoRolling();
+    } else {
+      setTimeout(() => {
+        this.rolling();
+        this.autoRolling();
+      }, 3000);
```

Comment:
여기서도 이미 시간 지연을 가능하게 해주는 `delay`함수와 async-await을 활용해볼 수 있겠네요~
```suggestion
await delay(3000);
this.rolling();
this.autoRolling();
```

Reply:
그리고 if-else절 모두에서 아래 메서드 호출이 중복으로 일어나고 있는데 쉽게 제거할 수 있을 것 같아요 :)
```javascript
this.rolling();
this.autoRolling();
```